ion/textwriter.go

```diff
@@ -372,14 +375,28 @@ func (w *textWriter) writeFieldName(api string) error {
 }
 
 // writeAnnotations writes out the annotations for a value.
-func (w *textWriter) writeAnnotations() error {
+func (w *textWriter) writeAnnotations(api string) error {
 	as := w.annotations
 	w.annotations = nil
 
-	for _, a := range as {
-		if err := writeSymbol(a, w.out); err != nil {
+	// If possible, resolve symbols as their names, else default to ID.
+	symbols := make([]string, len(as))
+	for i, a := range as {
+		id, _ := w.resolve(api, a)
+		name, hasName := w.lst.FindByID(id)
```

Comment:
I think I'm missing something here:

* We get the SID for the string `a` ("annotation"?)
* If it exists, we look it up in the table and are guaranteed to get back the same string value as `a`
* If it doesn't exist, we fall back to `a`.

It seems like we could skip the whole thing and just write `a`. What value is the symbol lookup getting us?